Mark evidence improvements

kelby-shelton opened the pull request:
Added support for results object

kelby-shelton committed:
Added support for results object

kelby-shelton committed:
Removed extra debug statement

philroyer-phantom commented on custom_functions/mark_evidence.py:
```diff
@@ -49,29 +37,67 @@ def mark_evidence(container=None, object_id=None, content_type=None, **kwargs):
+    # If content added is type 'action run',
+    # then iterate through an input object that is a results object,
+    # and append the action_run_id's to data
+    if isinstance(input_object, list) and content_type == 'actionrun':
```
Should this be content_type.lower() for consistency with the other places we check the content_type?

philroyer-phantom commented on custom_functions/mark_evidence.py:
```diff
@@ -1,10 +1,10 @@
     Args:
         container (CEF type: phantom container id): Container ID or Container Object
-        object_id (CEF type: *): Id of object to be added - artifact id, note id, etc.
+        input_object (CEF type: *): Id or object to be added - artifact id, note id, etc.
```
I think users will need a bit more here to make it clear how to use the `results` object. It is not clear that the `actionrun` type maps to keyword_arguments:results. Also, the results object will fail if there is another block between the action run and the custom function, including if there is a join block. Maybe something like this:

input_object (CEF type: *): The object to mark as evidence. This could be a vault_id, artifact_id, note_id, or if the previous playbook block is an action then "keyword_argument:results" can be used with the content_type "actionrun"

philroyer-phantom replied:
Also, I think container needs to be removed from the content_type options (I left it out above) because there seems to be no way to pass the literal string "container" into the custom function. It just ends up as a blank string.

philroyer-phantom commented on custom_functions/mark_evidence.py:
```diff
@@ -49,29 +37,67 @@ def mark_evidence(container=None, object_id=None, content_type=None, **kwargs):
+        # If data is still an empty list after for loop, 
+        # it indicates that the input_object was not a valid results object
+        if not data:
+            raise TypeError("The input for 'object_id' is not a valid integer or supported object.")
```
This should say "input_object" instead of "object_id", right?

philroyer-phantom reviewed (changes requested):
Suggested removing container type, using .lower() consistently, and improving actionrun description.

kelby-shelton committed:
Various documentation and input fixes

- Changed input_object description
- Changed reference to 'object_id'
- Changed 'container' object_type to 'container_id' as Phantom does not allow literal value of 'container'

philroyer-phantom commented on custom_functions/mark_evidence.json:
```diff
@@ -17,16 +17,16 @@
                 "*"
             ],
-            "description": "Id of object to be added - artifact id, note id, etc.",
+            "description": "The object to marked as evidence. This could be a vault_id, artifact_id, note_id, or if the previous playbook block is an action then \"keyword_argument:results\" can be used with the content_type \"actionrun\".",
```
Two things here:
1. "to marked as evidence" should be "to be marked as evidence"
2. container_id is back in, so please add it to the list

philroyer-phantom commented on custom_functions/mark_evidence.py:
```diff
@@ -1,46 +1,34 @@
     Args:
         container (CEF type: phantom container id): Container ID or Container Object
-        object_id (CEF type: *): Id of object to be added - artifact id, note id, etc.
+        input_object (CEF type: *): The object to marked as evidence. This could be a vault_id, artifact_id, note_id, or if the previous playbook block is an action then "keyword_argument:results" can be used with the content_type "actionrun".
```
(sorry, commented on the json above, but the comment should apply to the python)

Two things here:

"to marked as evidence" should be "to be marked as evidence"
container_id is back in, so please add it to the list

philroyer-phantom commented on custom_functions/mark_evidence.py:
```diff
@@ -1,46 +1,34 @@
         content_type (CEF type: *): The content type of the object to add as evidence which must be one of the following:
                         
                         vault_id
                         artifact
```
Actually I think it should be vault_id, artifact_id, container_id, note_id, and actionrun. That way it is more clear which is an ID and which is an object.

philroyer-phantom reviewed (changes requested):
Just one more quick change.

kelby-shelton committed:
Adjusted input object description and standardized object_type inputs

kelby-shelton committed:
Added vault hash description

philroyer-phantom commented on custom_functions/mark_evidence.py:
```diff
@@ -1,46 +1,34 @@
     Args:
         container (CEF type: phantom container id): Container ID or Container Object
-        object_id (CEF type: *): Id of object to be added - artifact id, note id, etc.
+        input_object (CEF type: *): The object to mark as evidence. This could be a vault_id, artifact_id, note_id, container_id, or actionrun_id. If the previous playbook block is an action then "keyword_argument:results" can be used for the actionrun_id with the content_type "actionrun". Vault_id can be an ID or a vault hash.
```
I don't actionrun_id is supported, so that should be changed to "actionrun object".

philroyer-phantom replied:
Now that the content type has been changed from actionrun to actionrun_id, could you please remove `with the content_type "actionrun"` from the end of the third sentence?

philroyer-phantom reviewed (changes requested):
One more minor change

kelby-shelton committed:
Added support for actionrun_id

Adjusted input to be actionrun_id for consistency and added support for actionrun_id

philroyer-phantom commented on custom_functions/mark_evidence.py:
```diff
@@ -49,29 +37,91 @@ def mark_evidence(container=None, object_id=None, content_type=None, **kwargs):
+    # and append the action_run_id's to data
+    if isinstance(input_object, list) and content_type.lower() == 'actionrun_id':
+        for action_result in input_object:
+            if action_result.get('action_run_id'):
```
Lines 45 and 48 are inconsistent now, with one being action_run_id and one being actionrun_id. It looks like action_run_id is the correct one so line 48 should be changed.

philroyer-phantom reviewed (changes requested):
Still a few inconsistencies around the action run id.

kelby-shelton committed:
Adjusted to action_run_id

Adjusted all mentions of actionrun or actionrun_id to action_run_id for internal consistency with the name of the key in the "results" object.

philroyer-phantom commented:
Looks good! Thank you Kelby!
